src/base/BaseV4Quoter.sol

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+import {IPoolManager} from "@uniswap/v4-core/src/interfaces/IPoolManager.sol";
+import {BalanceDelta} from "@uniswap/v4-core/src/types/BalanceDelta.sol";
+import {PoolKey} from "@uniswap/v4-core/src/types/PoolKey.sol";
+import {QuoterRevert} from "../libraries/QuoterRevert.sol";
+import {SqrtPriceLimitHelper} from "../libraries/SqrtPriceLimitHelper.sol";
+import {SafeCallback} from "../base/SafeCallback.sol";
+import {PoolId, PoolIdLibrary} from "@uniswap/v4-core/src/types/PoolId.sol";
+
+abstract contract BaseV4Quoter is SafeCallback {
```

Comment:
This base contract is used so that we can inherit it into siyu's MixedQuoter (that quotes V2, V3, and V4)